### PR TITLE
fix rustls-provider-example hpke no-std support

### PR DESCRIPTION
**Pull request by brodycj**
(with some very minor refactoring)

---

should be finally ready now ... apologies for so much noise ... looks like codecov is complaining about 2 lines of code for `no-std` since I don't think we can unit-test this; I don't think this should be an issue since CI does already verify that we can compile this for `no-std`

### CODE REVIEW

File: provider-example/src/hpke.rs [hunk 1]
```diff
@@ -209,18 +207,35 @@ impl HpkeOpener for HpkeRsReceiver {
...
+    }
 }
 
 #[cfg(not(feature = "std"))]
-fn other_err(err: impl Send + Sync + 'static) -> Error {
-    Error::General(alloc::format!("{}", err));
+mod error_util {
+    use core::fmt::Display;
+    use rustls::Error;
+    use rustls::OtherError;
```
Comment (brodycj) [inline provider-example/src/hpke.rs:233]:
```suggestion
```

File: provider-example/src/hpke.rs [hunk 2]
```diff
@@ -209,18 +207,35 @@ impl HpkeOpener for HpkeRsReceiver {
...
+
+    #[cfg(not(feature = "std"))]
+    pub(crate) fn wrap_err(err: impl Display + Send + Sync + 'static) -> Error {
```
Comment (brodycj) [inline provider-example/src/hpke.rs:236]:
```suggestion
    pub(crate) fn wrap_err(err: impl Display) -> Error {
```

**Review by brodycj — COMMENTED** [review]
should remove some things not needed for `no-std`

File: provider-example/src/hpke.rs [hunk 3]
```diff
@@ -213,14 +211,28 @@ impl HpkeOpener for HpkeRsReceiver {
...
+        Error::Other(OtherError(Arc::new(err)))
+    }
 }
 
 #[cfg(not(feature = "std"))]
-fn other_err(err: impl Send + Sync + 'static) -> Error {
-    Error::General(alloc::format!("{}", err));
+mod error_util {
+    use core::any::Any;
+    use rustls::Error;
```
Comment (brodycj) [inline provider-example/src/hpke.rs:231]: oops forgot this, apologies

```suggestion
    use rustls::Error;
    use rustls::OtherError;
```

**Review by djc — APPROVED** [review]
Nice!

**Pull request merged**